test-data/unit/fine-grained.test

```diff
@@ -10205,3 +10205,27 @@ C
 [builtins fixtures/dict.pyi]
 [out]
 ==
+
+[case testNamedTupleNestedCrash]
+import m
+[file m.py]
+from typing import NamedTuple
+
+class NT(NamedTuple):
+    class C: ...
+    x: int
+    y: int
+
+[file m.py.2]
+from typing import NamedTuple
+
+class NT(NamedTuple):
+    class C: ...
+    x: int
+    y: int
+# change
```

Comment:
Maybe also test a change in another module to test AST stripping and reprocessing of an invalid named tuple?

Reply:
Yeah, it actually doesn't work. This btw also affects the recent TypedDict PR too, I am going to fix both.